addons/point_of_sale/static/src/js/models.js

```diff
@@ -222,10 +222,46 @@ class PosGlobalState extends PosModel {
         this.db.add_partners(this.partners);
     }
     _loadProductProduct(products) {
+        const productMap = {};
+        const productTemplateMap = {};
+
+        function assignApplicableItems(pricelist, correspondingProduct, pricelistItem) {
```

Comment:
do not do inner function, do a normal private one

Reply:
due to your second comment/suggestion, the function didn't made sense anymore so it was removed.